src/thor/trace_route_action.cc

```diff
@@ -394,92 +395,105 @@ thor_worker_t::map_match(Api& request) {
 
     // trace_attributes always returns a single trip path and may have discontinuities
     if (options.action() == Options::trace_attributes) {
-      // we make a new route to hold the result of each iteraton of top k
+      // we make a new route to hold the result of each iteration of top k
       auto& route = *request.mutable_trip()->mutable_routes()->Add();
-      path_map_match(match_results, m_temp_path_edges, *route.mutable_legs()->Add(),
+      path_map_match(match_results, path_edges, *route.mutable_legs()->Add(),
                      m_temp_route_discontinuities);
     } // trace_route can return multiple trip paths
     else {
 
-      // here we break the path edges into groups of contiguous ones so that
+      // here we break the path edges and match results into contiguous sets so that
       // where they are discontinuous we can make them into separate routes
-      m_temp_disjoint_edge_groups = {{{m_temp_path_edges.front()}, {}}};
+      using edge_group_t =
```

Comment:
i completely changed how the disjoint edge groups work. previously they were copying the path edges into many sub lists. instead i store the first and last iterators of the range of path edges that are relevant avoiding any copies. additionally i store both the match results on either side of a discontinuity, in other words for each edge group we store the first and last path edge iterators and the first and last match result iterators.